### PR TITLE
clang: Fix hipstdpar test relying on default target

### PR DESCRIPTION
**Pull request by arsenm**
Use explicit target and stop restricting hosts it can run on.

### CODE REVIEW

File: clang/test/Driver/hipstdpar.c [hunk 1]
```diff
@@ -1,21 +1,17 @@
-// REQUIRES: x86-registered-target
-// REQUIRES: amdgpu-registered-target
-// REQUIRES: system-linux
```
Comment (shiltian) [inline clang/test/Driver/hipstdpar.c:3]: I'm not sure if the removal of `system-linux` would cause issue.

Comment (arsenm) [reply clang/test/Driver/hipstdpar.c:3]: This is a pile of workarounds, there's no reason any of these tests should be host dependent

**Comment by AlexVlx** [comment]
@arsenm what are you actually trying to fix and what do you expect this to do?

**Comment by arsenm** [comment]
> @arsenm what are you actually trying to fix and what do you expect this to do?

Fix not running tests except on linux. We should have maximum host test coverage, and this test has no reason to depend on the host. All it needs is the explicit target instead of relying on the default

**Comment by arsenm** [comment]
Window bot passed, which was the important bit. Linux failed on a different test entirely

**Review by perry-ca — APPROVED** [review]
This works on z/OS.  Thanks for fixing it.

**Pull request merged**